app/javascript/bundles/Groceries/components/Groceries.tsx

```diff
@@ -110,23 +102,75 @@ export default class Groceries extends React.Component<RouteComponentProps, Groc
           <Row className="justify-content-md-center" style={topPad}>
             <Col>
               <Form.Group>
-                <Form.Control size="lg" type="text" placeholder="Search for the groceries you need" />
+                <Form.Control
+                  size="lg"
+                  type="text"
+                  placeholder="Search for the groceries you need"
+                  onChange={(event: React.ChangeEvent<HTMLInputElement>): void => this.onSearch(event.target.value)}
+                />
               </Form.Group>
             </Col>
             <Col><h3 className="text-center">Basket</h3></Col>
           </Row>
           <Row>
             <Col>
               <GroceryList
-                items={this.state.itemsInList}
+                items={this.getItemsInList()}
                 handleAddButtonPressed={this.handleAddButtonPressed}
                 handleQuantityChange={this.handleQuantityChange}
               />
             </Col>
-            <Col><Basket items={this.state.itemsInCart} handleRemovedButtonPressed={this.handleRemoveButtonPressed} /></Col>
+            <Col>
+              <Basket
+                items={this.getItemsInCart()}
+                handleRemovedButtonPressed={this.handleRemoveButtonPressed}
+                handleQuantityChange={this.handleQuantityChange}
+              />
+            </Col>
           </Row>
         </Container>
       </>
     );
   }
+
+  getItemsInList(): GroceryLineItem[] {
+    return this.state.groceryLineItems.filter((item) => item.location == GroceryLineItemLocation.List)
+  }
+
+  getItemsInCart(): GroceryLineItem[] {
+    return this.state.groceryLineItems.filter((item) => item.location == GroceryLineItemLocation.Cart)
+  }
+
+  updateLineItemInList(lineItems: GroceryLineItem[], updatedItem: GroceryLineItem): GroceryLineItem[] {
+    return lineItems.map((item) => item.id == updatedItem.id ? updatedItem : item)
+  }
+
+  changePropertiesOfGroceryLineItem(id: string, properties: Partial<GroceryLineItem>, moveToEnd: boolean): void {
+    const groceryItem = this.state.groceryLineItems.find((item) => item.id === id) as GroceryLineItem
+
+    if (moveToEnd) {
```

Comment:
This decides on whether we want to put the item to the end of the list or not. E.g. updating quantity we don't. Updating location (cart vs list) we do. Maybe another reason to separate into two arrays...